testdata/result/expr/braced_new_constructor.sql.txt

```diff
@@ -0,0 +1,271 @@
+                    Name: &ast.Ident{
+                      NamePos: 300,
+                      NameEnd: 304,
+                      Name:    "name",
+                    },
+                    Value: &ast.BracedConstructorFieldValueExpr{
+                      Colon: 304,
+                      Expr:  &ast.StringLiteral{
+                        ValuePos: 306,
+                        ValueEnd: 315,
+                        Value:    "Scorpio",
+                      },
+                    },
+                  },
+                  &ast.BracedConstructorField{
+                    Name: &ast.Ident{
+                      NamePos: 320,
+                      NameEnd: 325,
+                      Name:    "index",
+                    },
+                    Value: &ast.BracedConstructorFieldValueExpr{
+                      Colon: 325,
+                      Expr:  &ast.IntLiteral{
+                        ValuePos: 327,
+                        ValueEnd: 328,
+                        Base:     10,
+                        Value:    "1",
+                      },
+                    },
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+      &ast.BracedConstructorField{
+        Name: &ast.Ident{
+          NamePos: 336,
+          NameEnd: 347,
+          Name:    "all_planets",
+        },
+        Value: &ast.BracedConstructorFieldValueExpr{
+          Colon: 347,
+          Expr:  &ast.ScalarSubQuery{
+            Lparen: 349,
+            Rparen: 378,
+            Query:  &ast.Select{
+              Select:   350,
+              Distinct: false,
+              AsStruct: false,
+              Results:  []ast.SelectItem{
+                &ast.ExprSelectItem{
+                  Expr: &ast.Ident{
+                    NamePos: 357,
+                    NameEnd: 364,
+                    Name:    "planets",
+                  },
+                },
+              },
+              From: &ast.From{
+                From:   365,
+                Source: &ast.TableName{
+                  Table: &ast.Ident{
+                    NamePos: 370,
+                    NameEnd: 378,
+                    Name:    "SolTable",
+                  },
+                  Hint:   (*ast.Hint)(nil),
+                  As:     (*ast.AsAlias)(nil),
+                  Sample: (*ast.TableSample)(nil),
+                },
+              },
+              Where:   (*ast.Where)(nil),
+              GroupBy: (*ast.GroupBy)(nil),
+              Having:  (*ast.Having)(nil),
+              OrderBy: (*ast.OrderBy)(nil),
+              Limit:   (*ast.Limit)(nil),
+            },
+          },
+        },
+      },
+    },
+  },
+}
+
+--- SQL
+NEW Universe {name: "Sol", closest_planets: ARRAY["Mercury", "Venus", "Earth"], star {radius_miles: 432690, age: 4603000000}, constellations: ARRAY[{name: "Libra", index: 0}, {name: "Scorpio", index: 1}], all_planets: (SELECT planets FROM SolTable)}
```

Comment:
I feel `[...]` to `ARRAY[...]` is not good, but I don't want to fix it in this PR.

Reply:
I made another issue and PR.
https://github.com/cloudspannerecosystem/memefish/issues/144
https://github.com/cloudspannerecosystem/memefish/pull/145

Reply:
Fixed 
https://github.com/cloudspannerecosystem/memefish/pull/119/commits/1dfdd8d1aa6fdf8f45a569694b17e9fbc55aef74